### ISSUE TITLE
⬆️ Track upgrade to cordova android@8.1.0 and ios@5.1.1

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-docs#519]
We have been working on a covid-19 tracking reference app in the
https://github.com/covid19database/phone-app fork.

There, we upgraded to the latest version of both `cordova-android` and `cordova-ios` modules, and to the latest versions of the plugins.
The https://github.com/e-mission/e-mission-phone/tree/upgrade_to_latest_cordova branch is tracking the changes required.

This is tracking the effort to port the changes back.

**Comment by shankari** [comment]
✨ Improvements ❇️ 
- latest cordova versions on both android and iOS
- latest versions of most other plugins
- supports kotlin on android and swift on iOS
- enables CI so we know whether the build is broken (https://github.com/e-mission/e-mission-phone/pull/639)
- WKWebView on iOS, all references to UIWebView have been removed
  - add native calls to aggregate functions to work around CORS issues (https://github.com/e-mission/e-mission-phone/pull/641/commits/06177b9ba1652cab54f9593597718d17dd56c7c2)

- basic functionality works
- parts which need to be ported/tested over the coming weeks, in order of priority. If anybody can help with testing, please let me know by 
  - devapp
  - test push notifications
  - port deploy code
  - debug and fix trip end notifications

**Comment by shankari** [comment]
Current status: 
- basic functionality works
- parts which need to be ported/tested over the coming weeks, in order of priority.
  - devapp
  - test push notifications
  - port deploy code
  - test openid auth
  - debug and fix trip end notifications

If you would like to contribute testing results, you can do so here.

**Comment by shankari** [comment]
> port deploy code

Tackling this first so that we can finish the javascript changes before upgrading the devapp.
The new deploy interface is:
https://github.com/ionic-team/cordova-plugin-ionic/blob/master/docs/interfaces/ideploypluginapi.md

Without the port, we get the error:

```
[Error] Unhandled Promise Rejection: TypeError: deploy.check is not a function. (In 'deploy.check(resolve, reject)', 'deploy.check' is undefined)
	(anonymous function) (updatecheck.js:52)
	promiseReactionJob
```

**Comment by shankari** [comment]
Deploy port done (https://github.com/e-mission/e-mission-phone/pull/641/commits/78503b99bbd199d7f63d741663b6bbbb84511ae4).
Next step: check that the push notifications work.

After that, I plan to work on the devapp and upgrade to the even more recent version of e-mission before fixing the trip end notification code since there may be additional changes required.

**Comment by shankari** [comment]
Push notifications work on both android and iOS.
On iOS, you need the hack to use APNS tokens.
https://github.com/e-mission/e-mission-docs/issues/437#issuecomment-513506146
until we change the server code as well

Given the extra configuration, it might be easier to just use the FCM token throughout and remove the mapping code from the server. But I am a bit leery about FCM lock-in, and being unable to switch to other providers in the future.

**Comment by shankari** [comment]
Ok, so I'm going to merge this, and then make the same changes to the devapp and the emTripLog app.
Open issues:
- trip end notifications: which I will address after I upgrade to cordova 9
- no testing of OpenIdAuth code: hopefully the FabMob team will do this and report

**Comment by shankari** [comment]
I'm going to declare this done right now, since I already closed #554
